jax/_src/api.py

```diff
@@ -521,10 +521,23 @@ def compile(self) -> 'Compiled':
         self._lowering.compile(), self.in_tree, self.out_tree,
         self.donate_argnums, self._no_kwargs)
 
-  def compiler_ir(self, dialect: Optional[str] = None):
+  def compiler_ir(self, dialect: Optional[str] = None, **kwargs):
```

Comment:
I'm increasingly leaning towards making `compiler_ir` return an `ir.Module` object. This is mostly because MLIR printing is not free. In particular, I recently ran across a case where the printed IR might be disproportionately large compared to the C++ object graph (callsiteloc).

This would have a downside that the `ir.Module` object may not agree between different Python extensions. But perhaps that isn't so bad and it could be covered by documentation?

Reply:
> This is mostly because MLIR printing is not free.

Something related, don't know if worth creating an issue, so I will just mention it here. Large constants are not hidden in MLIR printing, even a 10 by 10 matrix can bloat printed IR.

Reply:
@soraros is that causing a noticeable problem for you? If so, file a bug?

Reply:
I'm +1 on returning a Module. Code that crosses boundaries already needs to be aware of such things.

binary=True builds into a bytes based IO. For tool interop, the additional python str conversion is pure overhead (shouldn't change the size at rest much but python has to make multiple passes/copies in order to determine and use the minimal encoded string representation).